### ISSUE TITLE
Uncaught exception when doing crime

### ISSUE THREAD

**Issue by davad** [bitburner-official/bitburner-src#732]
Currently in BN2.1 using Bitburner v2.4.0.

Doing any crime in the slums shows this error in the debug menu:

```
Uncaught TypeError: r.J5.focusPenalty is not a function
    at p.commit (CrimeWork.ts:58:26)
    at p.process (CrimeWork.ts:40:12)
    at w.a [as processWork] (PlayerObjectWorkMethods.ts:13:37)
    at Object.updateGame (engine.tsx:91:5)
    at start (engine.tsx:397:14)
```

The progress bar hangs at 100% and no reward is given.

**Comment by myCatsName** [comment]
I'm not able to get that error. I start bitnode 2.1, slums, try some crimes by script or by hand - the progress bar works, player overview increases stats, and money increases when crime succeeds.  I am on a different version though.

Here are a few lines that are probably related.

CrimeWork.ts line 58
![image](https://github.com/bitburner-official/bitburner-src/assets/141260118/d4f42e8f-2269-4eeb-8f48-79b25fd85e3c)


PlayerObjectGeneralMethods.ts line 1218
![image](https://github.com/bitburner-official/bitburner-src/assets/141260118/76aee4eb-1dd7-4c11-a5a1-8e66808085c7)
imo focusPenalty() returns a value, which is called and used as a focusPenalty const in the the calculation of gains each time a crime timer has finished. "Returning a function vs returning the result of the function" mood, but it seems to work properly...

Not to doubt the issue, but it's not immediately clear why the problem, or how to repeat it.

**Comment by davad** [comment]
I found and fixed the problem. It was an issue with my save file.

I stepped through the code and found a hardcoded `1` instead of a function.  I exported the save file, base64 decoded it, and I found  `focusPenalty` property in the PlayerSave section.

`cat prettified.json| jq '.data.PlayerSave | fromjson | .data`
<img width="467" alt="Screenshot 2023-08-12 at 12 52 05 PM" src="https://github.com/bitburner-official/bitburner-src/assets/190387/2bb473c1-ec91-4a22-ace6-6133257a4273">

Removing `focusPenalty` and `focus` from my save file, then encoding and importing the save fixed the issue.

I have _no_ idea how those ended up in my save file, though. Maybe when I was mucking around to get SF -1 achievements?
